compiler/rustc_const_eval/src/errors.rs

```diff
@@ -425,7 +425,7 @@ pub struct ValidationFailure {
     #[primary_span]
     pub span: Span,
     #[note(const_eval_validation_failure_note)]
-    pub ub_note: Option<()>,
+    pub ub_note: (),
```

Comment:
There's probably something else that should be done instead of a `()` field here...? I don't know enough about our diagnostics stuff, unfortunately.

Reply:
`()` is fine. You could put `#[note]` on the struct itself but it changes the ordering of the labels/notes.